docs/architecture/architecture.tex

```diff
@@ -123,6 +123,16 @@ \subsection{Programming languages and auxiliary programs}
 For continuous integration and version control we use Git and Github. Locally we use Maven to build the project. On GitHub we use Travis CI to verify the builds.
 We also use the JavaFX Scene Builder as a tool to design parts of the GUI because this provides a better preview of what the interface will look like while creating it.
 
+\subsection{Data organization}
+The data that we read from the file is stored in datatables containing columns of specific types. These datatables together form the model on which we can perform the analyses. The values are stored as datavalues inside columns having the same type as the datavalue. These datavalues are divided in subclasses for strings, integers, floats, dates and booleans.
+
+Date values are a bit more complex than, for example an integer. Date values can vary in being a timestamp, period of time, just a date or just a time value. The section for dates of the datavalue structure has been implemented to have multiple types of values for these varying types of dates.
+
+To refer to data inside methods, we use identifiers. These identifiers can point to a column or table by finding the corresponding data for its given string which is the name of the searched data. The software also has a describer implemented, which can be used to refer to proceses like a computation or a single value in a column of a row. 
+
+In week 4.8 we found an issue that reading data from a large excel file takes up a lot of time and memory. We thought this could have been caused by the fact that we used hashmaps to store the rows and the iteration over its entries could take up a lot of time. We tried using linked hashmaps instead but that did not improve the reading speed very much. The conclusion to which we came was was that the POI library uses a lot of memory when reading over large excel files.
```

Comment:
je hebt het hier eerst over geheugen, dan noem je er een oplossing voor en zeg je dat de oplossing niet werkte want het maakte het programma niet sneller, maar de snelheid was niet het probleem die je noemde. 
Je moet in dist stukje naar voren laten komen dat we voor de columns hashmaps gebruiken en dat we gekeken hebben of gebruikmaken van linkedlist (geen linked hashmaps) minder geheugen gebruikt (met name voor grote bestanden), onze conclusie was dat het niet veel uitmaakt en dat i.v.m. de snelheid van hashmaps we voor de hashmap hebben gekozen. 
Verder het inlezen was niet het grote probleem (inlezen kost ook veel geheugen, maar is niet problematisch), maar het verwerken, daar zagen we namelijk dat er enorm veel geheugen werd gebruikt, dat de garbage collector druk bezig was en dat ie daardoor erg traag was. Dit hebben we opgelost. 

Dus je moet zorgen dat de kern van dit stukje is dat we bewust voor de hashmaps kozen. De reden waarom we dit onderzochte, mag je noemen, maar moet niet het belangrijkste uit dit stuk zijn.